File: testdata/bench/javascript/BenchmarkTest00029.js
// Vulnerable: XSS via variable indirection through res.send
const express = require('express');
const router = express.Router();
router.get('/test029', (req, res) => {
    const input = req.query.msg;
    const message = input;
    const output = '<div>' + message + '</div>';
    res.send(output);
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00029.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;
use std::process::Command;

// Vulnerable: Command injection via variable indirection
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("input").unwrap_or(&String::new()).clone();
    let x = param.clone();
    let output = Command::new("sh").arg("-c").arg(&x).output().expect("failed");
    HttpResponse::Ok().body(String::from_utf8_lossy(&output.stdout).to_string())
}
